src/main/java/com/faforever/api/email/MailBodyBuilder.java

```diff
@@ -0,0 +1,145 @@
+package com.faforever.api.email;
+
+import com.faforever.api.config.FafApiProperties;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.text.MessageFormat.format;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class MailBodyBuilder implements InitializingBean {
+
+  @Getter
+  public enum Template {
```

Comment:
I am just thinking about using abstract class as a template with constructor with a path to template or direct string of template and method format(Map<> variables), and have each mail type as subclass. This class would do all the validation. But I am too tired to sort out all the details :)

Reply:
Yeah. Sounds like a use case for sealed classes.

Reply:
So after looking at it I see not better solution

Moving away from enum to sealed classes as typed data carriers (carrying the variables) we lose the ability to iterate over the existing list of classes without reflection + we'd need to instantiate them to have template validation on startup.
Having each class implement a format method leads to code duplication unless you abstract it but then you end up with the same method as of now.

The only differences in the templates are: 
* different configuration in the properties (which imho shouldn't be coded into a POJO - doesn't make sense you either need to inject the properties or pass them in the method)
* different variables: but we abstracted this nicely with the public methods of the MailBodyBuilder methods

Using more classes felt like doing OOP for the sake of OOP without adding benefits. I'm open to improvement PRs though if you find a better solution.